docs/adr/0008-preview-deployment.md

```diff
@@ -0,0 +1,80 @@
+# Preview Deployment
+
+- Status: proposed
+- Deciders: Monitor team
+- Date: 2024-05-08
+
+Technical Story: [MNTOR-3110](https://mozilla-hub.atlassian.net/browse/MNTOR-3110)
+
+## Context and Problem Statement
+
+With the first phase of CI/CD in place, at present, all changes merged into `main` are immediately deployed to the staging environment, but we are unable to continuously deploy to production because we cannot release changesets until they have been validated by QA, accessibility (a11y), and other teams.
+
+Pull requests can be categorized into several types:
+
+- Dependency updates (primarily through Dependabot)
+- Feature enhancements and bug fixes that require QA verification
+- Bug fixes that do not necessitate QA action
+- Accessibility fixes that need approval from the a11y team
+
+It seems counterproductive to delay all deployments due to a small number of changesets requiring verification. Moreover, the impact of postponing a11y changes is uncertain.
+
+If we stick to our current release process, the steps would be:
+
+- Establish a release branch
+- Undo the unwanted changes (although cherry-picking the desired changes might be more appropriate in this case)
+- Tag the release branch
+- Deploy the tag to the staging environment, undoing the unwanted changes (which could interfere with the verification process)
+- Deploy to the production environment
+- Revert the staging environment to the main branch
+
+This approach appears to be disruptive and prone to errors. If our test automation and feature flags are not yet reliable enough to ensure safe continuous deployment, we should consider how to enable PR verification by QA/a11y/etc. prior to merging, similar to the process we use with Netlify and Storybook for components.
```

Comment:
I don't know that it's only because test automation and flags aren't reliable, there are going to be some set of changes that are risky but can't be behind a flag. An example would be a large dependency update that requires a rewrite for instance like next-auth 5.

Reply:
Hopefully that set is small but I think it's not the null set ;)